check user and repo for redirects when using git via SSH transport

koalajoe23 opened the pull request:
fixes #30565 

When using git with a gitea hosted repository, the HTTP-Transport did honor the user and repository redirects, which are created when renaming a user or repo and also when transferring ownership of a repo to a different organization. This is extremely helpful, as repo URLs remain stable and do not have to be migrated on each client's worktree and other places, e.g. CI at once.

The SSH transport - which I favor - did not know of these redirections and I implemented a lookup during the `serv` command.

**Considerations**

- I am not a golang dev, feel free to suggest a more idiomatic implementation
- This introduces DB lookups during the `serv` command. I am not sure how gitea handles this and if this may lead to performance or concurrency problems. My manual tests do not indicate any, but I'd be glad if an experienced contributor could verify this.
- I haven't written any test cases yet. I'm a bit tired now but I try to commit them at a later time.
- I did manual tests to verify correct handling of user rename and repo transfer though, also in combination.

lunny reviewed (changes requested):
All operations should be performed under private routers, rather than directly in the serv command.

koalajoe23 commented:
> All operations should be performed under private routers, rather than directly in the serv command.

I see, much cleaner to implement it there.

Moved the redirection handling there and also cleaned it up a bit

koalajoe23 commented:
Apart from missing tests for the SSH redirects, Git LFS over SSH tests now fail. Not sure why though...
